File: 251229/비오는 날/rainy-day.py
class Weather:
    def __init__(self, date, day, weather):
        self.date = date
        self.day = day
        self.weather = weather


n = int(input())
arr = [tuple(input().split()) for _ in range(n)]
weathers = [Weather(date, day, w) for date, day, w in arr]

target_idx = -1
for i, w in enumerate(weathers):
    if w.weather == 'Rain':
        if target_idx == -1:
            target_idx = i
        else:
            if arr[target_idx][0] >= w.date:
                target_idx = i

ans = weathers[target_idx]
print(f'{ans.date} {ans.day} {ans.weather}')

File: 251229/비오는 날/rainy-day.js
class Weather {
    constructor(date, day, weather) {
        this.date = date;
        this.day = day;
        this.weather = weather;
    }
}

const fs = require('fs');
const input = fs.readFileSync(0).toString().trim().split('\n');
const n = Number(input[0]);

let ans = new Weather('9999-99-99', '', '');
for (let i = 1; i <= n; i++) {
    const [date, day, weather] = input[i].split(' ');

    if (weather === 'Rain') {
        if (ans.date >= date) {
            ans = {...ans, date, day, weather};
        }
    }
}

console.log(`${ans.date} ${ans.day} ${ans.weather}`);